documentation/modules/ROOT/partials/modules/mysql-connector/con-mysql-connector-events.adoc

```diff
@@ -3,15 +3,67 @@
 [id="mysql-connector-events_{context}"]
 = MySQL connector events
 
-All data change events produced by the {prodname} MySQL connector contain a key and a value. The change event key and the change event value each contain a _schema_ and a _payload_ where the schema describes the structure of the payload and the payload contains the data.
+The {prodname} MySQL connector generates a data change event for each row-level `INSERT`, `UPDATE`, and `DELETE` operation. Each event contains a key and a value. The structure of the key and the value depends on the table that was changed. 
 
-WARNING: The MySQL connector ensures that all Kafka Connect schema names adhere to the link:http://avro.apache.org/docs/current/spec.html#names[Avro schema name format]. This is important as any character that is not a latin letter or underscore is replaced by an underscore which can lead to unexpected conflicts in schema names when the logical server names, database names, and table names container other characters that are replaced with these underscores.
+{prodname} and Kafka Connect are designed around _continuous streams of event messages_. However, the structure of these events may change over time, which can be difficult for consumers to handle. To address this, each event contains the schema for its content, which makes each event self-contained. The following skeleton JSON shows the basic structure of a change event: 
```

Comment:
> each event contains the schema for its content

That's not 100% correct, if you think of Avro and the schema registry. In that case, events will only contain the schema id, which a consumer can use to obtain the schema from the registry. Perhaps say "either inline or by means of a schema registry..."?

Reply:
I changed it to this (will be in the next commit): 
To address this, each event contains the schema for its content or, if you are using a schema registry, a schema ID that a consumer can use to obtain the schema from the registry. This makes each event self-contained.